add try / except to address issue #10154 to_datetime, Inconsistent be…

vincentdavis opened the pull request:
closes #10154 to_datetime, Inconsistent behavior with invalid dates.

results are now

```
In [2]: pd.to_datetime('2015-02-29', format="%Y-%m-%d", coerce=True)
Out[2]: NaT

In [3]: pd.to_datetime('2015-03-32', format="%Y-%m-%d", coerce=True)
Out[3]: NaT

In [4]: pd.to_datetime('2015-02-32', format="%Y-%m-%d", coerce=True)
Out[4]: NaT

In [5]: pd.to_datetime('2015-04-31', format="%Y-%m-%d", coerce=True)
Out[5]: NaT
```

Commit:
support both sqlalchemy engines and connections Fixes #7877

update pymysql to 0.6.3 to avoid cursor bugs

Add documentation and tests for SQLAlchemy connectables

explicit reference to connection/engine in docs

Temporary table test

pass compile the connectable

vincentdavis committed:
add try / except to address issue #10154 to_datetime, Inconsistent behavior with invalid dates.

vincentdavis committed:
add try / except to address issue #10154 to_datetime, Inconsistent behavior with invalid dates.

jreback commented:
needs tests

vincentdavis committed:
Add test for issue #10154

vincentdavis committed:
Add test for issue #10154

jreback commented on pandas/tests/test_tseries.py:
```diff
@@ -737,6 +737,13 @@ def test_get_period_field_raises_on_out_of_range(self):
         self.assertRaises(ValueError, period.get_period_field_arr, -1, np.empty(1), 0)
 
+class TestDaysInMonth(tm.TestCase):
+    def test_day_not_in_month_coerce_true(self):
```
pls add the issue number as a comment

vincentdavis replied:
I thought I did, not sure what went wrong.

jreback commented on pandas/tests/test_tseries.py:
```diff
@@ -737,6 +737,13 @@ def test_get_period_field_raises_on_out_of_range(self):
+        self.assertTrue(isnull(to_datetime('2015-02-29', format="%Y-%m-%d", coerce=True)))
+        self.assertTrue(isnull(to_datetime('2015-02-32', format="%Y-%m-%d", coerce=True)))
+        self.assertTrue(isnull(to_datetime('2015-04-31', format="%Y-%m-%d", coerce=True)))
+
```
can you add off these cases again, but with ``coerce=False`` (and assert that they raise ``ValueError``)

vincentdavis committed:
Include test for ValueError for fix of issue #10154

vincentdavis committed:
Include test for ValueError for fix of issue #10154

jreback commented on pandas/tests/test_tseries.py:
```diff
@@ -737,6 +737,17 @@ def test_get_period_field_raises_on_out_of_range(self):
+        self.assertTrue(isnull(to_datetime('2015-02-32', format="%Y-%m-%d", coerce=True)))
+        self.assertTrue(isnull(to_datetime('2015-04-31', format="%Y-%m-%d", coerce=True)))
+    def test_day_not_in_month_coerce_false(self):
+        self.assertRaises(ValueError, to_datetime, '2015-02-29', format="%Y-%m-%d", coerce=False)
```
yeh, test both ``errors='ignore'`` and ``errors='raise'`` here

vincentdavis replied:
This looks correct
```
In [10]: pd.to_datetime('2015-02-29', errors='ignore',  coerce=False)
Out[10]: '2015-02-29'
```
I would expect the same from this but get a ValueError.
```
In [12]: pd.to_datetime('2015-02-29', errors='ignore', format="%Y-%m-%d", coerce=False)
....
ValueError: day is out of range for month
```
adding format="%Y-%m-%d" should not change the output.
Is this correct and also needs to be fixed?

jreback commented:
can you rebase

vincentdavis commented:
On Fri, Jun 26, 2015 at 5:39 PM, jreback <notifications@github.com> wrote:

> can you rebase


I will work on that this weekend.


Vincent Davis
720-301-3003

vincentdavis committed:
Add more to_datetime tests

vincentdavis committed:
Add more to_datetime tests

Commit:
Updated to_hdf doc string

Updated the docstring to make clear that a HDFStore is required for a buffer, if a path is not directly passed

jorisvandenbossche committed:
Merge pull request #10105 from graingert/sqlalchemy-connectable

support both sqlalchemy engines and connections

jreback committed:
Merge pull request #10499 from joshlk/master

Updated to_hdf doc string

sinhrks committed:
Merge pull request #10500 from sinhrks/test_quantile

TST: DataFrame.quantile should have Float64Index

vincentdavis committed:
rebase

vincentdavis committed:
fix merge
